### PR TITLE
build: upgrade Go 1.17 and k8s.io v0.23.5

### PR DESCRIPTION
**Pull request by xcoulon**
also, remove openshift-ci/Dockerfile.tools since it's not used anymore.

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,32 +6,120 @@ require (
...
+	k8s.io/api v0.24.2
+	k8s.io/apimachinery v0.24.2
+	k8s.io/client-go v0.24.2
+	k8s.io/kubectl v0.24.2
+	k8s.io/metrics v0.24.2
+	sigs.k8s.io/controller-runtime v0.12.1
 )
 
-go 1.16
+require (
```
Comment (mfrancisc) [inline go.mod:35]: quite a few additional dependencies with the upgrade!? :)

File: go.mod [hunk 2]
```diff
@@ -6,32 +6,120 @@ require (
...
+	k8s.io/component-base v0.24.2 // indirect
+	k8s.io/klog/v2 v2.60.1 // indirect
+	k8s.io/kube-openapi v0.0.0-20220328201542-3ee0da9b0b42 // indirect
+	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9 // indirect
+	sigs.k8s.io/json v0.0.0-20211208200746-9f7c6b3444d2 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
+	sigs.k8s.io/yaml v1.3.0 // indirect
+)
+
+replace github.com/codeready-toolchain/api => github.com/xcoulon/api v0.0.0-20220928143841-49954da92052
```
Comment (mfrancisc) [inline go.mod:121]: I guess this will be removed once the merges will be ready right?

**Review by mfrancisc — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (xcoulon) [reply go.mod:35]: yes, `go mod tidy` generated a lot of changes in `go.mod` 🤷‍♂️

File: go.mod (hunk 2 above)
Comment (xcoulon) [reply go.mod:121]: yes, and we have a job called `ci-check-gomod / go.mod replacements` which looks for such replacements, so we don't forget about reverting them once the PR for github.com/codeready-toolchain/api has been merged

File: .github/workflows/publish-operators-for-e2e-tests.yml [hunk 3]
```diff
@@ -64,12 +64,12 @@ jobs:
...
 
     - name: Prepare tools
-      uses: codeready-toolchain/toolchain-cicd/prepare-tools-action@master
+      uses: xcoulon/toolchain-cicd/prepare-tools-action@d426c7b
 
     # Execute composite action using values from PR event
     # Is executed only for PR events - in that case the comment field is empty
     - name: Publish current operator bundles for host & member based on PR event
-      uses: codeready-toolchain/toolchain-cicd/publish-operators-for-e2e-tests@master
+      uses: xcoulon/toolchain-cicd/publish-operators-for-e2e-tests@d426c7b
```
Comment (MatousJobanek) [inline .github/workflows/publish-operators-for-e2e-tests.yml:72]: Let's revert these changes

File: go.mod [hunk 4]
```diff
@@ -6,32 +6,119 @@ require (
...
+	k8s.io/kube-openapi v0.0.0-20211115234752-e816edb12b65 // indirect
+	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9 // indirect
+	sigs.k8s.io/json v0.0.0-20211208200746-9f7c6b3444d2 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
+	sigs.k8s.io/yaml v1.3.0 // indirect
+)
+
+replace github.com/codeready-toolchain/api => github.com/xcoulon/api v0.0.0-20220930084836-ecf3e69a0460
+
+replace github.com/codeready-toolchain/toolchain-common => github.com/xcoulon/toolchain-common v0.0.0-20220929132149-50a438c62f49
```
Comment (MatousJobanek) [inline go.mod:122]: reminder

File: make/run-cicd-script.mk [hunk 5]
```diff
@@ -1,5 +1,5 @@
 
-OWNER_AND_BRANCH_LOCATION=codeready-toolchain/toolchain-cicd/master
+OWNER_AND_BRANCH_LOCATION=xcoulon/toolchain-cicd/d426c7b
```
Comment (MatousJobanek) [inline make/run-cicd-script.mk:2]: and this one too

**Review by MatousJobanek — APPROVED** [review]

File: go.mod (hunk 4 above)
Comment (xcoulon) [reply go.mod]: addressed in 35861c2

File: make/run-cicd-script.mk (hunk 5 above)
Comment (xcoulon) [reply make/run-cicd-script.mk]: reverted in https://github.com/codeready-toolchain/toolchain-e2e/commit/7e01e2fe1f9159f8a27001e50f1c5e881390ad10

File: .github/workflows/publish-operators-for-e2e-tests.yml (hunk 3 above)
Comment (xcoulon) [reply .github/workflows/publish-operators-for-e2e-tests.yml]: reverted in https://github.com/codeready-toolchain/toolchain-e2e/commit/7e01e2fe1f9159f8a27001e50f1c5e881390ad10

**Comment by xcoulon** [comment]
as agreed with @MatousJobanek, I'm bypassing the branch protection for this very particular case, because some of the e2e tests are still running with Go 1.16 which is not compatible with the dependencies we now have.
